packages/frontend-shared/src/components/BaseLink.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+  <!-- eslint-disable vue/multiline-html-element-content-newline  -->
+  <a
+    :href="props.href"
+    :class="classes"
+  ><slot /></a>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+
+export default defineComponent({
+  inheritAttrs: true,
+})
+</script>
+
+<script setup lang="ts">
+/* eslint-disable no-duplicate-imports */
+
+import { computed, useAttrs } from 'vue'
+
+import type { AnchorHTMLAttributes } from 'vue'
+
+const attrs = useAttrs() as AnchorHTMLAttributes
+
+// if no class is used, the link receives a default color
+const classes = computed(() => {
+  const hocus = props.useDefaultHocus ? 'hocus-link-default' : ''
+  const color = hasTailwindTextColorClass(attrs.class) ? '' : 'text-indigo-500'
```

Comment:
I could use input on this implementation. I want a way to use a default color class in this component & have it get out of the way when the parent passes a conflicting class, since without doing this we have 2 `!important` rules on the component targeting the same property with different values, which would have unwanted results.

The alternative here is to drive the color with props like `color` or `variant` but I think it's natural to add colors in the same way we do elsewhere, with classes, and expect a class on a link to just work.

Reply:
Amusingly, I think this is what unocss is trying to solve. We're not going to switch to using it, but this is the first time where I find myself wanting it.

Reply:
```scss
<style lang="scss" scoped>
a {
  color: theme('colors.indigo.500')
}
</style>
```

Reply:
This allows us to demote the !important and let the user win.

Reply:
I'm totally fine with this. I think there are lots more places where this is useful, if we are comfortable opting out of the utility classes for setting defaults. It's not a bad way to divide up styles that we are intentionally baking in to the component vs defaults that can and should be overridden easily. Which I think will mostly be colors anyway.